docs/ext/commands/slash_commands.rst

```diff
@@ -146,6 +146,58 @@ All the other types may be converted implicitly, similarly to :ref:`ext_commands
   \*\* Role and Member may be used together to create a "mentionable" (:class:`Union[Role, Member]`)
 
 
+Number Ranges
++++++++++++++
+
+:class:`int` and :class:`float` parameters support minimum and maximum allowed
+values using the ``lt``, ``le``, ``gt``, ``ge`` parameters on :func:`Param <ext.commands.Param>`.
+For instance, you could restrict an option to only accept positive integers:
+
+.. code-block:: python3
+
+    @bot.slash_command()
+    async def command(
+        inter: disnake.ApplicationCommandInteraction,
+        amount: int = commands.Param(gt=0),
+    ):
+        ...
+
+
+Instead of using :func:`Param <ext.commands.Param>`, you can also use a :class:`Range` annotation.
+The range bounds are both inclusive; the type of the option is determined by the range bounds,
+with the option being a :class:`float` if at least one of the bounds is a :class:`float`, and :class:`int` otherwise.
+
+.. code-block:: python3
+
+    @bot.slash_command()
+    async def ranges(
+        inter: disnake.ApplicationCommandInteraction,
+        a: int = commands.Param(lt=0),  # negative int
+        b: commands.Range[1, ...],      # positive int
+        c: commands.Range[0, 10],       # 0 - 10 int
+        d: commands.Range[0, 10.0],     # 0 - 10 float
+    ):
```

Comment:
This won't work, the parameter with `Param` would have to be at the end.
```suggestion
    async def ranges(
        inter: disnake.ApplicationCommandInteraction,
        a: commands.Range[0, 10],       # 0 - 10 int
        b: commands.Range[0, 10.0],     # 0 - 10 float
        c: int = commands.Param(lt=0),  # negative int
        d: commands.Range[1, ...],      # positive int
    ):
```

Reply:
this still isn't at the end

Reply:
good catch, thanks
Now that I think about it... `commands.Param` should probably be removed entirely there, since the code is meant to showcase `commands.Range`

Reply:
fixed in 370b148c9386f15b5bf809d1dd876c4bda834041